homeassistant/components/devialet/media_player.py

```diff
@@ -0,0 +1,274 @@
+        """Return the device info."""
+        return DeviceInfo(
+            identifiers={(DOMAIN, self._serial)},
+            name=self._name,
+            manufacturer=MANUFACTURER,
+            model=self._client.model,
+            sw_version=self._client.version,
+        )
+
+    @property
+    def name(self) -> str:
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique id of the device."""
+        return self._serial
+
+    @property
+    def state(self) -> MediaPlayerState | None:
+        """Return the state of the device."""
+        if not self._client.is_available:
+            return MediaPlayerState.OFF
+
+        playing_state = self._client.playing_state
+
+        if not playing_state:
+            return MediaPlayerState.IDLE
+        if playing_state == "playing":
+            return MediaPlayerState.PLAYING
+        if playing_state == "paused":
+            return MediaPlayerState.PAUSED
+        return MediaPlayerState.ON
+
+    @property
+    def volume_level(self) -> float | None:
+        """Volume level of the media player (0..1)."""
+        return self._client.volume_level
+
+    @property
+    def is_volume_muted(self) -> bool | None:
+        """Return boolean if volume is currently muted."""
+        return self._client.is_volume_muted
+
+    @property
+    def source_list(self) -> list[str] | None:
+        """Return the list of available input sources."""
+        return self._client.source_list
+
+    @property
+    def sound_mode_list(self) -> list[str] | None:
+        """Return the list of available sound modes."""
+        return sorted(SOUND_MODES)
+
+    @property
+    def media_artist(self) -> str | None:
+        """Artist of current playing media, music track only."""
+        return self._client.media_artist
+
+    @property
+    def media_album_name(self) -> str | None:
+        """Album name of current playing media, music track only."""
+        return self._client.media_album_name
+
+    @property
+    def media_title(self) -> str | None:
+        """Return the current media info."""
+        if not self._client.media_title:
+            return self.source
+
+        return self._client.media_title
+
+    @property
+    def media_image_url(self) -> str | None:
+        """Image url of current playing media."""
+        return self._client.media_image_url
+
+    @property
+    def media_duration(self) -> int | None:
+        """Duration of current playing media in seconds."""
+        return self._client.media_duration
+
+    @property
+    def media_position(self) -> int | None:
+        """Position of current playing media in seconds."""
+        return self._client.current_position
+
+    @property
+    def media_position_updated_at(self) -> datetime.datetime | None:
+        """When was the position of the current playing media valid."""
+        return self._client.position_updated_at
```

Comment:
These can also be switched to the shorthand instance attributes

Reply:
Same as above. Thanks for your review by the way!